fms/modules/attention.py

```diff
@@ -690,7 +815,18 @@ def forward(
         else:
             keys_compute, values_compute = keys, values
 
-        if attn_compute_dict["is_prefill"](**attn_kwargs):
+        updated_attn_kwargs: Union[AttentionKwargs, SinkAttentionKwargs]
+
+        if self.has_sinks:
+            updated_attn_kwargs = {
+                "attn_name": "sdpa_with_sinks",
+                "sinks": self.sinks,
+                "sliding_window": sliding_window,
+            }
```

Comment:
the right to do this would be for the headless model itself to change its own attn_name to sdpa_with_sinks, and then, here, if attn_name is already the right one and has_sinks is True, the sinks and sliding_window parameters are added to attn_kwargs, instead of fully modifying whatever the user has passed in. You can see it in Bert/Roberta, where the original attn_name is changed if needed